web/src/main/java/org/springframework/security/web/server/authentication/ott/GenerateOneTimeTokenWebFilter.java

```diff
@@ -0,0 +1,85 @@
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.web.server.authentication.ott;
+
+import reactor.core.publisher.Mono;
+
+import org.springframework.http.HttpMethod;
+import org.springframework.http.server.reactive.ServerHttpRequest;
+import org.springframework.security.authentication.ott.GenerateOneTimeTokenRequest;
+import org.springframework.security.authentication.ott.reactive.ReactiveOneTimeTokenService;
+import org.springframework.security.web.server.util.matcher.ServerWebExchangeMatcher;
+import org.springframework.security.web.server.util.matcher.ServerWebExchangeMatchers;
+import org.springframework.util.Assert;
+import org.springframework.util.StringUtils;
+import org.springframework.web.server.ServerWebExchange;
+import org.springframework.web.server.WebFilter;
+import org.springframework.web.server.WebFilterChain;
+
+/**
+ * {@link WebFilter} implementation that process a One-Time Token generation request.
+ *
+ * @author Max Batischev
+ * @since 6.4
+ * @see ReactiveOneTimeTokenService
+ */
+public final class GenerateOneTimeTokenWebFilter implements WebFilter {
+
+	private final ReactiveOneTimeTokenService oneTimeTokenService;
+
+	private ServerWebExchangeMatcher matcher = ServerWebExchangeMatchers.pathMatchers(HttpMethod.POST, "/ott/generate");
+
+	private final ServerGeneratedOneTimeTokenHandler generatedOneTimeTokenHandler;
+
+	public GenerateOneTimeTokenWebFilter(ReactiveOneTimeTokenService oneTimeTokenService,
+			ServerGeneratedOneTimeTokenHandler generatedOneTimeTokenHandler) {
+		Assert.notNull(generatedOneTimeTokenHandler, "generatedOneTimeTokenHandler cannot be null");
+		Assert.notNull(oneTimeTokenService, "oneTimeTokenService cannot be null");
+		this.generatedOneTimeTokenHandler = generatedOneTimeTokenHandler;
+		this.oneTimeTokenService = oneTimeTokenService;
+	}
+
+	@Override
+	public Mono<Void> filter(ServerWebExchange exchange, WebFilterChain chain) {
+		return this.matcher.matches(exchange)
+			.filter(ServerWebExchangeMatcher.MatchResult::isMatch)
+			.switchIfEmpty(chain.filter(exchange).then(Mono.empty()))
+			.flatMap((matchResult) -> resolveUsername(exchange.getRequest()))
+			.switchIfEmpty(chain.filter(exchange).then(Mono.empty()))
+			.map(GenerateOneTimeTokenRequest::new)
+			.flatMap(this.oneTimeTokenService::generate)
+			.flatMap((token) -> this.generatedOneTimeTokenHandler.handle(exchange, token));
+	}
+
+	private Mono<String> resolveUsername(ServerHttpRequest request) {
+		String username = request.getQueryParams().getFirst("username");
```

Comment:
I can suggest adding the `allowFormEncodedBodyParameter` property. In the servlet part, the logic for converting the request into h is built on the query parameter “token”. It will be strange if the servlet and reactive parts are very different, especially since the servlet part is already in the main branch.